index.bs

```diff
@@ -4422,6 +4483,203 @@ Of the extended attributes defined in this specification, only the [{{Exposed}}]
+
+<div algorithm>
+  The <dfn for=module>url</dfn> of a [=module=] |module| is given by the following steps:
+
+    1.  Let |specifier| be |module|'s [=module/specifier=].
+    1.  Return the [=concatenation=] of « "<code>import:std:</code>", |specifier| ».
+</div>
+
+Note that like [=namespaces=], modules do not create types.
+
+Rather than defining behavior when accessed, the [=attributes=] of a [=module=] act as a simple
+storage location. The relevant language binding defines the steps to "set a module attribute" to
+an IDL value of the type of the [=attribute=].
+
+Note: See the ECMAScript binding's definition of [=set a module attribute=].
+
+Each module may have <dfn for=module>evaluation steps</dfn> which are executed when the module is
+first imported.
+These steps have access to a <dfn>this module</dfn> value, which is the [=module=] being
+evaluated, and a [=Realm=].
+They may throw an [=exception=], which prevents the module from being accessed.
+The [=module/evaluation steps=] for a module |m| must invoke the "set a module attribute"
+algorithm to initialize each [=attribute=] in |m|'s [=module/module members=].
+
+<div class="example" noexport>
+
+    The following [=IDL fragment=] defines a [=module=] exposing a function to get the current
+    timezone, which returns an instance of an interface exposed in the same module.
+
+    <pre class="syntax">
+        [Exposed=Window, SecureContext]
+        module <dfn interface>temporal</dfn> {
+          interface <dfn interface>Timezone</dfn> {
+            readonly attribute USVString name;
+            long long offsetMs(long long unixTime);
+          };
+          temporal.Timezone getCurrentTimezone();
+          readonly attribute temporal.Timezone <dfn attribute for=temporal>initialTimezone</dfn>;
+        };
+    </pre>
+
+    <blockquote>
+        The [=module/evaluation steps=] of the {{temporal}} module in |realm| are as follows:
+            1.  Let |initial| be a [=new=] {{Timezone}} in |realm|, representing the User
+                Agent's preferred time zone.
+            1.  [=set a module attribute|Set the attribute=] {{temporal/initialTimezone}} of
+                <b>this module</b> to |initial|.
+    </blockquote>
+
+    An ECMAScript implementation would then expose a <code class="idl">"std:temporal"</code>
+    module which includes an [=interface object=] for the interface and a function for the operation:
+
+    <pre highlight="js">
+        import * from "std:temporal" as temporal;
+        Object.keys(temporal);                      // Evaluates to ["Timezone", "getCurrentTimezone", "currentTimezone"]
+        typeof temporal.Timezone;                   // Evaluates to "function"
+        typeof temporal.getCurrentTimezone;         // Evaluates to "function"
+        typeof temporal.initialTimezone;            // Evaluates to "object"
+    </pre>
+</div>
+
+### Syntax ### {#idl-modules-syntax}
```

Comment:
I note that nothing else has syntax subsections, although I do agree it's a nice thematic break here. Not sure what to do. (Although it doesn't matter much.)

Reply:
If I ever get time, I'd like to see if this split makes sense for the other sections as well, but that's still a while off.